bigtable/integration_test.go

```diff
@@ -96,7 +100,48 @@ func TestMain(m *testing.M) {
 			c.Cluster, "us-central1-b", "1",
 		)
 	}
-	os.Exit(m.Run())
+	exit := m.Run()
+	if err := cleanup(c); err != nil {
+		log.Printf("Post-test cleanup failed: %v", err)
+	}
+	os.Exit(exit)
+}
+
+func cleanup(c IntegrationTestConfig) error {
+	// Cleanup resources marked with bt-it-. Check timestamp, remove over 7D old.
+
+	instancePrefix := "bt-it-"
+	if !c.UseProd {
+		return nil
+	}
+	ctx := context.Background()
+	var opts []option.ClientOption
+	iac, err := NewInstanceAdminClient(ctx, c.Project, opts...)
+	if err != nil {
+		return err
+	}
+	instances, err := iac.Instances(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, info := range instances {
+		if strings.HasPrefix(info.Name, instancePrefix) {
+			timestamp := info.Name[len(instancePrefix):]
+			t, err := strconv.ParseInt(timestamp, 10, 64)
+			if err != nil {
+				return err
+			}
+			uT := time.Unix(t, 0)
+			sevenDays := time.Hour * 24 * 7
+			if time.Now().After(uT.Add(sevenDays)) {
+				// delete an instance older than 7 days
+				iac.DeleteInstance(ctx, info.Name)
```

Comment:
I think even a shorter timeframe than 7 days is probably good given how expensive bigtable instances are, no? Also, is it just instances that need to be deleted, or any other resources as well?

Reply:
There are instances and clusters, but they are coupled, so 99% sure I don't need to also call delete cluster. @kolea2 can confirm likely.

Reply:
yep! https://cloud.google.com/bigtable/docs/deleting-instance